asio/asio-config.cmake.in

```diff
@@ -0,0 +1,7 @@
+@PACKAGE_INIT@
+
+include(${CMAKE_CURRENT_LIST_DIR}/asio-targets.cmake)
+check_required_components(asio)
+
+include(CMakeFindDependencyMacro)
+find_dependency(OpenSSL QUIET)
```

Comment:
This is wrong, you must find dependencies before you make the targets available. The purpose of `find_dependency` is to forward arguments from the calling `find_package` command as well. This file should just be:

```cmake
include(CMakeFindDependencyMacro)
find_dependency(OpenSSL)

include("${CMAKE_CURRENT_LIST_DIR}/asioTargets.cmake")
```

See my other comment about htis file not requiring configuration.